### PR TITLE
Mobile improvements

### PR DESCRIPTION
**Pull request by justmytwospence**
DO NOT MERGE yet

This adds images and a few other small fixes to the mobile documentation. There's currently 2 problems with the images:

1 . My phone adds a weird white background when screenshotting so it washes out some of the code blocks in a weird way.
2. Because mobile screenshots are so tall and narrow, they end up taking a whole screen of scroll-space which is hard to read. They need to have a fixed height. If we could put them to the side and wrap the text around them, that would be even better.

### CODE REVIEW

**Comment by justmytwospence** [comment]
I've reduced the size of all the images for now:

```sh
for file in *.png; do convert -resize x500 $file ${file%.png}-small.png; done
```

File: docs/mobile/authoring.md [hunk 1]
```diff
@@ -29,8 +29,8 @@ expand snippets, and to write a snippet expansion for different exercise
 types. Bonus points if your IDE of choice allows you to automatically generate
 UUIDs for your exercises.
```
Comment (BGordts) [inline docs/mobile/authoring.md]: Maybe add 'We'll explain later what UUIDS are and what you need them for

File: docs/mobile/development.md [hunk 2]
```diff
@@ -79,12 +79,12 @@ mobile-teach preview /path/to/lesson-file.yml
 ```
```
Comment (BGordts) [inline docs/mobile/development.md]: Note for the text before: should we mention what NPM is and how to install it?

File: docs/mobile/development.md [hunk 3]
```diff
@@ -79,12 +79,12 @@ mobile-teach preview /path/to/lesson-file.yml
...
 
 You can then open a web browser (we recommend Firefox version 58 or later) to
-`http://localhost:8080`. Open the browser tools (`Command-Option-i` for Firefox)
-and click the phone icon to simulate a phone view port. You can even specify
-which phone you want to simulate (eg iPhone, Nexus, Galaxy, etc) by clicking the
-bar at the top of the viewport and selecting a device model. We recommend
-previewing on the smallest phone supported by DataCamp for Mobile - the iPhone
-SE.
+[`localhost:8080`](http://localhost:8080). Open the browser tools
+(`Command-Option-i` for Firefox) and click the phone icon to simulate a phone
```
Comment (BGordts) [inline docs/mobile/development.md]: Maybe we should also mention how it works on Chrome?

File: docs/mobile/development.md [hunk 4]
```diff
@@ -79,12 +79,12 @@ mobile-teach preview /path/to/lesson-file.yml
...
+view port. You can even specify which phone you want to simulate (eg iPhone,
+Nexus, Galaxy, etc) by clicking the bar at the top of the viewport and selecting
+a device model. We recommend previewing on the smallest phone supported by
```
Comment (BGordts) [inline docs/mobile/development.md]: I would not say that, recommend rather to test on an iPhone 6 or so, as this is a more 'default' screen with then an iPhone 5. I think we should optimize for our biggest portion of users, not the most constrained ones.

File: docs/mobile/exercises/README.md [hunk 5]
```diff
@@ -3,14 +3,14 @@
...
-| [Select Output](select-output.md)     | What is the output of this code? Console output         |
-| [Select Table](select-table.md)       | What is the output of this code? Formatted table output |
-| [Select Code](select-code.md)         | Select the code to return the output.                   |
-| [Tap](tap.md)                         | Complete the code to return the output.                 |
-| [Multiple Choice](multiple-choice.md) | Answer the multiple choice question.                    |
-| [Reorder](reorder.md)                 | Reorder the code to return the output.                  |
+| Type                                                    | Description                                             |
+| :---                                                    | :----------                                             |
+| [Select Output](/mobile/exercises/select-output.md)     | What is the output of this code? Console output         |
+| [Select Table](/mobile/exercises/select-table.md)       | What is the output of this code? Formatted table output |
```
Comment (BGordts) [inline docs/mobile/exercises/README.md]: Maybe mention that tables are mainly / only used for SQL

File: docs/mobile/exercises/README.md [hunk 6]
```diff
@@ -3,14 +3,14 @@
...
+| [Select Code](/mobile/exercises/select-code.md)         | Select the code to return the output.                   |
+| [Tap](/mobile/exercises/tap.md)                         | Complete the code to return the output.                 |
+| [Multiple Choice](/mobile/exercises/multiple-choice.md) | Answer the multiple choice question.                    |
+| [Reorder](/mobile/exercises/reorder.md)                 | Reorder the code to return the output.                  |
 
 ...but they all share a similar structure. In general, you can think of an
```
Comment (BGordts) [inline docs/mobile/exercises/README.md]: I would add a screenshot for the exercise below, so people can get an idea at first glance about context, question, code...

File: docs/mobile/exercises/README.md [hunk 7]
```diff
@@ -170,36 +170,27 @@ image: assets/images/my-cool-plot.png
...
-context:
-  text: Study the table below to answer the following question.
-  table: |-
-    foo, bar, baz
-    1  ,2   ,3
-    4  ,5   ,6
-```
+![Example context image](/images/mobile/image-context-small.png)
 
 ## feedback
```
Comment (BGordts) [inline docs/mobile/exercises/README.md]: Don't mention this maybe? I thought we decided we would not longer support it?

Also, how it works right now is that feedback is ALWAYS shown, UNLESS overwritten by feedback_wrong or distractors.

Comment (BGordts) [reply docs/mobile/exercises/README.md]: So if an instructor only provides feedback and not feedback_wrong or distractors, and the user answers incorrectly, he will see the message provided in feedback (which might be 'great, well done!' :p)

File: docs/mobile/exercises/README.md [hunk 8]
```diff
@@ -231,11 +224,19 @@ tabs:
...
 
+An example of a `dataset` rendered as a `table`:
+
+![A tab from the exercise screen](/images/mobile/tabs-dataset-table-small.png) ![Opened dataset tab](/images/mobile/tabs-dataset-table-open-small.png)
+
+An example of a rendered `file` tab:
+
+![Opened file tab](/images/mobile/tabs-file-open-small.png)
+
 # Summary of available blocks
```
Comment (BGordts) [inline docs/mobile/exercises/README.md]: For the table:
- In a Multiple Choice question, the question is REQUIRED

File: docs/mobile/development.md (hunk 2 above)
Comment (justmytwospence) [reply docs/mobile/development.md]: I currently link to the download and installation page, which should hopefully cover installation, but yes I can add a quick explanation of what npm is.

File: docs/mobile/exercises/README.md (hunk 7 above)
Comment (justmytwospence) [reply docs/mobile/exercises/README.md]: Seems like that behavior would never be desired, but yes this is unsupported for now and I forgot to remove it.

File: docs/mobile/exercises/README.md (hunk 6 above)
Comment (justmytwospence) [reply docs/mobile/exercises/README.md]: right

File: docs/mobile/exercises/README.md (hunk 5 above)
Comment (justmytwospence) [reply docs/mobile/exercises/README.md]: right

File: docs/mobile/exercises/README.md (hunk 8 above)
Comment (justmytwospence) [reply docs/mobile/exercises/README.md]: right

**Pull request merged**